calcom/cal.com
# chore: makes production build ready for docker

## What does this PR do?

This PR modifies how you can use environment variables on runtime to have "one build, many deployments". With this update, you can now build Docker images, so anybody can deploy with their own environments.

This fix allows PR [#313](https://github.com/calcom/docker/pull/313) to work correctly.

Fixes [#12032](https://github.com/calcom/cal.com/issues/12032)

## Requirement/Documentation

- No extra requirements needed

## Type of change


## How should this be tested?

- Follow README.md instructions to build for production
- The repository may have some build Issues that should be attended to fix some types of errors, meanwhile, you can run `DISABLE_ESLINT_PLUGIN=true NODE_ENV=production CI=true yarn build` to avoid those errors
- Run `yarn start` to test
